plugins/repository-s3/src/test/java/org/elasticsearch/repositories/s3/S3BlobStoreRepositoryTests.java

```diff
@@ -71,21 +68,19 @@ public void wipeRepository() {
     }
 
     @Override
-    protected void createTestRepository(final String name, boolean verify) {
-        assertAcked(client().admin().cluster().preparePutRepository(name)
-            .setType(S3Repository.TYPE)
-            .setVerify(verify)
-            .setSettings(Settings.builder()
-                .put(S3Repository.BUCKET_SETTING.getKey(), bucket)
-                .put(S3Repository.BUFFER_SIZE_SETTING.getKey(), bufferSize)
-                .put(S3Repository.SERVER_SIDE_ENCRYPTION_SETTING.getKey(), serverSideEncryption)
-                .put(S3Repository.CANNED_ACL_SETTING.getKey(), cannedACL)
-                .put(S3Repository.STORAGE_CLASS_SETTING.getKey(), storageClass)));
+    protected String repositoryType() {
+        return S3Repository.TYPE;
     }
 
     @Override
-    protected void afterCreationCheck(Repository repository) {
-        assertThat(repository, instanceOf(S3Repository.class));
+    protected Settings repositorySettings() {
+        return Settings.builder()
+            .put(S3Repository.BUCKET_SETTING.getKey(), bucket)
+            .put(S3Repository.BUFFER_SIZE_SETTING.getKey(), bufferSize)
+            .put(S3Repository.SERVER_SIDE_ENCRYPTION_SETTING.getKey(), serverSideEncryption)
+            .put(S3Repository.CANNED_ACL_SETTING.getKey(), cannedACL)
+            .put(S3Repository.STORAGE_CLASS_SETTING.getKey(), storageClass)
```

Comment:
NIT: We seem to randomize `chunk_size` and `compress` for the Google and GS repository but not for the S3 one ... maybe just add that setting randomization to the abstract parent to dry things up and cover S3 as well?

Reply:
Good point, I pushed 235ef53291e5b05608396a2880d8814d5210e240